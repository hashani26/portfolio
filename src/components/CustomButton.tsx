import React from "react";
import { Button } from "antd";
import { ButtonProps } from "antd/lib/button";

export const CustomButton = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => (
    <>
      <Button data-testid="button" ref={ref} {...props}>
        {props.children}
      </Button>
    </>
  )
);
