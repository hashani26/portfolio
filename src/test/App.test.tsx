import React from 'react';
import { CustomButton } from '../components/CustomButton';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Button } from 'antd';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CustomButton></CustomButton>, div);
});

it('renders button correctly', () => {
  const { getByTestId } = render(<CustomButton>To Do</CustomButton>);
  expect(getByTestId('button')).toHaveTextContent('To Do');
});

it('matches snapshot 1', () => {
  const tree = renderer.create(<Button>Add Task</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('matches snapshot 2', () => {
  const tree = renderer.create(<Button>To Do</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
