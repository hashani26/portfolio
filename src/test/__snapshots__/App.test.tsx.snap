// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <span>
    Add Task
  </span>
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <span>
    To Do
  </span>
</button>
`;
